SalesforceCommerceCloud/pwa-kit
# @W-14338017@ Fix: accessibility focus for my account pages and promo code

# Description



This PR fixes several focus accessibility issues:
- When entering invalid promo code, error message is now read when focusing on the input field
    - This works on v2 but not on v3 due to a documented regression where there is no feedback provided when entering an invalid promo code
- Navigating through the my account pages (account details, wishlist, order history, addresses) sets the focus onto the corresponding page heading instead of staying on the navigation link
- after removing a product from wishlist, focus is set to `Wishlist` heading instead of onto the body
- Opening up an address form causes focus to shift to the address form
- Closing out of address form by saving or cancelling changes focus to the edit button that opened the address form
- removing an address changes focus to the `Addresses` heading on the Addresses page
- Canceling out of address form or removing address in checkout page changes focus to the `Shipping Addresses` heading

# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- (change1)

# How to Test-Drive This PR

- (step1)
1. `git checkout ju/fix-focus.W-14338017`
2. `npm ci`
3. `cd packages/template-retail-react-app`
4. `npm run start`
5. navigate to my account pages, specifically addresses
6. observe that focus shifts automatically to the heading of the corresponding page
7. observe that hitting the adding/editting address buttons sets focus to address form
8. observe that canceling or saving address form sets focus to address heading

<img width="352" alt="Screenshot 2023-12-20 at 10 48 03 AM" src="https://github.com/SalesforceCommerceCloud/pwa-kit/assets/88680517/56aad795-49a3-46be-a533-2da6531d4813">



# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
